doc/rest-api.md

```diff
@@ -2424,3 +2427,236 @@ Input (none at present):
             }
         }
     }
+## `/1.0/storage-pools`
+### GET
+ * Description: list of storage pools
+ * Introduced: with API extension `storage`
+ * Authentication: trusted
+ * Operation: sync
+ * Return: list of storage pools that are currently defined on the host
+
+    [
+        "/1.0/storage-pools/default",
+        "/1.0/storage-pools/pool1"
+        "/1.0/storage-pools/pool2"
+        "/1.0/storage-pools/pool3"
+        "/1.0/storage-pools/pool4"
+    ]
+
+### POST
+ * Description: create a new storage pool
+ * Introduced: with API extension `storage`
+ * Authentication: trusted
+ * Operation: sync
+ * Return: standard return value or standard error
+
+Input:
+
+    {
+        "config": {
+            "size": "10GB"
+        },
+        "driver": "zfs",
+        "name": "pool1"
+    }
+
+## `/1.0/cluster`
+### GET (optional `?password=<trust-password>`)
+ * Description: information about a cluster (such as networks and storage pools)
+ * Introduced: with API extension `clustering`
+ * Authentication: trusted or untrusted
+ * Operation: sync
+ * Return: dict representing a cluster
+
+    {
+        "type": "sync",
+        "status": "Success",
+        "status_code": 200,
+        "operation": "",
+        "error_code": 0,
+        "error": "",
+        "metadata": {
```

Comment:
So I think I'd expect something like:

```json
            "enabled": true,
            "status": "RUNNING",
            "current_server": "lxd01",
            "member_config": [
                {
                     "name": "storage-pools.default.config.source",
                     "description": "\"source\" property for storage pool \"default\""
                },
                {
                     "name": "networks.lxdbr0.config.bridge.external_interfaces",
                     "description": "\"bridge.external_interfaces\" property for network \"lxdbr0\""
                }
           ]
```

Reply:
I don't understand, the stuff listed above should be the only thing we actually need in the local database, everything else is in the global one. So when joining, ask for those things, put them in the local database and now when you merge the global and local bits for that storage or network, you've got everything you need.

Reply:
As discussed earlier, lets keep this for a further update of the API where we add a "nice" way to join the cluster. Until then, we'll just be using GetNetworks, GetStoragePools on the cluster side and CreateNetwork and CreateStoragePool on the local side.